internal/services/netapp/netapp_volume_helper.go

```diff
@@ -0,0 +1,970 @@
+}
+
+type ExportPolicyRule struct {
+	RuleIndex         int    `tfschema:"rule_index"`
+	AllowedClients    string `tfschema:"allowed_clients"`
+	Nfsv3Enabled      bool   `tfschema:"nfsv3_enabled"`
+	Nfsv41Enabled     bool   `tfschema:"nfsv41_enabled"`
+	UnixReadOnly      bool   `tfschema:"unix_read_only"`
+	UnixReadWrite     bool   `tfschema:"unix_read_write"`
+	RootAccessEnabled bool   `tfschema:"root_access_enabled"`
+}
+
+type DataProtectionReplication struct {
+	EndpointType           string `tfschema:"endpoint_type"`
+	RemoteVolumeLocation   string `tfschema:"remote_volume_location"`
+	RemoteVolumeResourceId string `tfschema:"remote_volume_resource_id"`
+	ReplicationFrequency   string `tfschema:"replication_frequency"`
+}
+
+type DataProtectionSnapshotPolicy struct {
+	DataProtectionSnapshotPolicy string `tfschema:"snapshot_policy_id"`
+}
+
+type VolumeSpecName string
+
+const (
+	VolumeSpecNameData       VolumeSpecName = "data"
+	VolumeSpecNameLog        VolumeSpecName = "log"
+	VolumeSpecNameShared     VolumeSpecName = "shared"
+	VolumeSpecNameDataBackup VolumeSpecName = "data-backup"
+	VolumeSpecNameLogBackup  VolumeSpecName = "log-backup"
+)
+
+func PossibleValuesForVolumeSpecName() []string {
+	return []string{
+		string(VolumeSpecNameData),
+		string(VolumeSpecNameLog),
+		string(VolumeSpecNameShared),
+		string(VolumeSpecNameDataBackup),
+		string(VolumeSpecNameLogBackup),
+	}
+}
+
+func RequiredVolumesForSAPHANA() []string {
+	return []string{
+		string(VolumeSpecNameData),
+		string(VolumeSpecNameLog),
+	}
+}
+
+type ProtocolType string
+
+const (
+	ProtocolTypeNfsV41 ProtocolType = "NFSv4.1"
+	ProtocolTypeNfsV3  ProtocolType = "NFSv3"
+	ProtocolTypeCifs   ProtocolType = "CIFS"
+)
+
+func PossibleValuesForProtocolType() []string {
+	return []string{
+		string(ProtocolTypeNfsV41),
+		string(ProtocolTypeNfsV3),
+		string(ProtocolTypeCifs),
+	}
+}
+
+func PossibleValuesForProtocolTypeAvg() []string {
+	return []string{
+		string(ProtocolTypeNfsV41),
+		string(ProtocolTypeNfsV3),
+	}
+}
+
+type ReplicationSchedule string
+
+const (
+	ReplicationSchedule10Minutes ReplicationSchedule = "10minutes"
+	ReplicationScheduleDaily     ReplicationSchedule = "daily"
+	ReplicationScheduleHourly    ReplicationSchedule = "hourly"
+)
+
+func PossibleValuesForReplicationSchedule() []string {
+	return []string{
+		string(ReplicationSchedule10Minutes),
+		string(ReplicationScheduleDaily),
+		string(ReplicationScheduleHourly),
+	}
+}
+
+// Diverging from the SDK volumegroups.SecurityStyle since it is defined as lower case
+// but the backend changes it to Pascal case on GET. Please refer to https://github.com/Azure/azure-sdk-for-go/issues/14684
+type SecurityStyle string
+
+const (
+	SecurityStyleUnix SecurityStyle = "Unix"
+	SecurityStyleNtfs SecurityStyle = "Ntfs"
+)
+
+func PossibleValuesForSecurityStyle() []string {
+	return []string{
+		string(SecurityStyleUnix),
```

Comment:
No, that's intentional, Ntfs security style is not supported for AVG, only NFS + Unix security style.

Reply:
I think it might still be missing here though? 

Just a few lines up you define `SecurityStyleNtfs` and also use have code for it [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/21290/files#diff-8bd6ceade72cb11afa90ab5d6b05874c1de990097d7e21a5517204026acc95cdR247) but a user can't set it because the only value available is `SecurityStyleUnix` because it's the only value in `PossibleValuesForSecurityStyle`

Reply:
I define that since I won't need to use strings to represent "Ntfs" down in some underlying validations.

Reply:
That makes sense but we can never hit that underlying validation because a user can never set it as `Ntfs` because it isn't in `PossibleValuesForSecurityStyle`. 

<img width="789" alt="image" src="https://user-images.githubusercontent.com/6515518/231524312-7b43a4ec-d060-4e3c-9086-7e50e534ed54.png">

`security_style` is using `PossibleValuesForSecurityStyle` and `PossibleValuesForSecurityStyle` only has `Unix`

<img width="393" alt="image" src="https://user-images.githubusercontent.com/6515518/231524520-fe3e4695-a2b7-414a-8b31-cbc07e8cd91f.png">

but in `expandNetAppVolumeGroupVolumes`, you're doing a validation check for `Ntfs` but that code will never be run as `Ntfs` can't be passed in by the user per `PossibleValuesForSecurityStyle`

<img width="599" alt="image" src="https://user-images.githubusercontent.com/6515518/231524826-fa13e6b1-ba05-40db-a771-cc8c0fa09554.png">

Reply:
I added as extra check, if you think I should remove it, I can. What worries me is some day in the future someone thinks that the schema has a bug and add the ntfs and provide a PR without full knowledge of how AVG works and you and I don't catch it, if we leave the validation, it will be a second layer of check.

Reply:
Removed this check and the previous one that checks if unix style is being applied to CIFS volume and that situation won't happen as well for AVG.